web/src/main/java/com/epam/esm/controllers/CertificateController.java

```diff
@@ -0,0 +1,198 @@
+     * <p> POST /web/certificates
+     * <p> Content type: application/json
+     *
+     * @param certificate {@link Certificate} object to be added
+     * @return ResponseEntity. Response status: 201 Created
+     */
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity createCertificate(
+            @RequestBody Certificate certificate
+    ) {
+
+        certificateService.create(certificate);
+        return new ResponseEntity(HttpStatus.CREATED);
+    }
+
+    /**
+     * Updates existing record in data source with corresponding info
+     * <p> PATCH /web/certificates
+     * <p> Content type: application/json
+     *
+     * @param certificate {@link Certificate} object to be updated
+     * @return <li>ResponseEntity. Response status: 204 No Content </li>
+     * <li>ResponseEntity. Response status: 404 Not Found. When no certificate with this id were found</li>
+     */
+    @PatchMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity updateCertificate(
+            @RequestBody Certificate certificate
+    ) {
+        certificateService.update(certificate);
+        return new ResponseEntity(HttpStatus.NO_CONTENT);
+    }
+
+    /**
+     * Deletes existing record in data source
+     * <p> DELETE /web/certificates/{id}
+     *
+     * @param id id of {@link Certificate} object to be deleted
+     * @return <li>ResponseEntity. Response status: 204 No Content </li>
+     * <li>ResponseEntity. Response status: 404 Not Found. When no certificate with this id was found</li>
+     */
+    @DeleteMapping("/{id}")
+    public ResponseEntity deleteCertificate(
+            @PathVariable String id
+    ) {
+        if (!isIdValid(id)) {
+            throw new ResponseException(new ValidationErrorMessage(ID_ERROR_MESSAGE, id, ERROR_DETAILS));
+        }
+        certificateService.delete(Long.parseLong(id.trim()));
+        return new ResponseEntity(HttpStatus.NO_CONTENT);
+    }
+
+    /**
+     * Returns  single record from data source with corresponding id
+     * <p> GET /web/certificates/{id}
+     *
+     * @param id id of {@link Certificate} object to be returned
+     * @return <li>ResponseEntity. Response status: 400 OK </li>
+     * <li>ResponseEntity. Response status: 404 Not Found. When no certificate with this id were found</li>
+     */
+    @GetMapping(value = "/{id}")
+    public ResponseEntity getById(@PathVariable("id") String id) {
+
+        if (!isIdValid(id)) {
+            throw new ResponseException(new ValidationErrorMessage(ID_ERROR_MESSAGE, id, ERROR_DETAILS));
+        }
+        return ResponseEntity.ok(certificateService.getById(Long.parseLong(id.trim())));
+    }
+
+    /**
+     * Handles exception and returns ErrorInfo object
+     *
+     * @return ErrorInfo object containing exception info with errorCode = 50002. Response status: 500 Internal Server Error.
+     */
+
+    private boolean isIdValid(String id) {
```

Comment:
@KukharchukDaniil All logic should be moved to service layer.

Reply:
fixed